### ISSUE TITLE
Builder problem: ssh_exchange_identification: Connection closed by remote host

### ISSUE THREAD

**Issue by rvadim** [deis/deis#4495]
Issue for builder from deis version 1.10.0:
Builder working normally(as expected) during ~20 hours, then users which try to push getting error:
```bash
$ git push deis master
ssh_exchange_identification: Connection closed by remote host
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
I don't actually testing, but i thing builder stop posting it host and port to etcd. Or maybe it posting only once for 24 hour TTL. So router cannot route requests to builder.

**Comment by rvadim** [comment]
Restart of builder helps, but next issue is:
```bash
$ git push deis master
fatal: 'my-app.git' does not appear to be a git repository
fatal: Could not read from remote repository.Please make sure you have the correct access rights
and the repository exists.
```
And i solve this by:
```bash
 $ nse deis-builder
 $ cd /home/git
 $ rm -R my-app.git
```

**Comment by rvadim** [comment]
Yes. Host and port no more published.
```bash
etcdctl ls /deis/builder/
/deis/builder/sshHostecdsaKey
/deis/builder/sshHosted25519Key
/deis/builder/users
/deis/builder/sshHostrsaPubKey
/deis/builder/sshHostrsaKey
/deis/builder/sshHostdsaKey
```

**Comment by rvadim** [comment]
And error is
```
Sep 16 09:25:50 dev-node-0.novalocal sh[29179]: [error] Etcd error setting host/port: Tried and failed 2 cluster connections: Unexpected HTTP status code
```
And problem is here:
https://github.com/deis/deis/blob/v1.10.0/builder/etcd/etcd.go#L361

**Comment by geeksoul-me** [comment]
I'm facing the same issue, this happens approximated every 24 hours. Restarting builder fixes this, but it keeps recurring.

```deisctl journal builder``` shows this
```
 [error] Etcd error setting host/port: Tried and failed 2 cluster connections: Unexpected HTTP status code
```

**Comment by technosophos** [comment]
Sorry, I just now saw this issue.

The cause of this is etcd not answering requests. We have to refresh the etcd host/port mappings frequently or else they expire out of etcd. When they expire, SSH traffic can no longer be routed from the edge of the cluster to the builder, and then all pushes to builder fail.

I know @mboersma is working on some fixes to etcd right now. I'll consult him about this, but I am wondering if raising the TTL might help, if this is really connection related.

I'm also gradually replacing the go-etcd client with the new official etcd client, but I suspect that will not land until Deis v2.

**Comment by rvadim** [comment]
I change break to continue on 16 September and builder still works great in my cluster with 30+ apps. I actually can't understand why this blocker issue does't fixed in 1.10.

**Comment by rvadim** [comment]
There is one problem: 
deis-builder fleet unit have not ExecStop handler, so when VM with builder have network problems for 5 - 10 min, you can get 2 builders in cluster.
In fact, with #4677 builder will never stopped automatically.
